app/views/orders/_service_desc.html.haml

```diff
@@ -19,5 +19,5 @@
     - if order_detail.stored_files.template_result.empty?
       = link_to 'Upload Order Form', order_order_detail_order_file_path(order_detail.order, order_detail), :class => 'btn'
     - else
-      = link_to('Uploaded Order Form', order_detail.stored_files.template_result.first.file.url)
+      = link_to "Uploaded Order Form", order_detail_first_template_result_path(order_detail)
```

Comment:
I moved `template_results` and `sample_results` to the `OrderDetailsController` which should avoid both the 404s and 403s on these.